content/blog/2020-09-28-september-20-community-gems.md

```diff
@@ -0,0 +1,198 @@
+directory in a remote location. For example, a pipeline stage could depend on a
+`model.pkl` file in your S3 bucket:
+
+```dvc
+$ dvc run -d s3://mybucket/model.pkl
+```
+
+DVC connects to the remote location (here, your S3 bucket) to check whether or
+not the indicated file (`model.pkl`) has changed since the last time `dvc repro`
+was run, but that's it. There's no caching involved.
+
+## CML questions
+
+### [Q: How can I use CML with my own Docker container?](https://discordapp.com/channels/485586884165107732/728693131557732403/757553135840526376)
+
+In many of our CML docs and videos, we've shown how to get CML on your CI
+(continuous integration) runner via a Docker container that comes with
+everything installed. But this is not the only way to use CML, especially if you
+want workflows to run in your own Docker container.
+
+You can install CML via `npm`, either in your own Docker container or in your CI
+workflow (i.e., in your GitHub Actions `.yaml` or GitLab CI `.yml` workflow
+file).
+
+To install CML as a package, you'll want to run:
+
+```bash
+$ npm i -g @dvcorg/cml
+```
+
+Note that you may need to install additional dependencies if you want to use DVC
+plots and Vega-Lite commands:
+
+```bash
+$ sudo apt-get install -y libcairo2-dev libpango1.0-dev libjpeg-dev libgif-dev \
+          librsvg2-dev libfontconfig-dev
+$ npm install -g vega-cli vega-lite
+```
+
+If you're installing CML as part of your workflow, you may need to install Node
+first-
+[check out our docs](https://github.com/iterative/cml#install-cml-as-a-package)
+for how to do this in GitHub Actions and GitLab CI.
+
+### [Q: After running a GitHub Action workflow that runs a DVC pipeline, I want to save the output of the pipeline. Why doesn't CML automatically save the output?](https://discordapp.com/channels/485586884165107732/728693131557732403/757686601953312988)
+
+What you're talking about sounds like an auto-commit: a `git commit` at the end
+of your CI workflow to commit any new artifacts to your Git repository. However,
+auto-commits have a lot of downsides- they don't make sense for a lot of users,
+and generally, it's better to re-create outputs as needed than save them forever
+in your Git repo.
+
+We created the DVC `run-cache` in part
+[to solve this issue](https://stackoverflow.com/questions/61245284/is-it-necessary-to-commit-dvc-files-from-our-ci-pipelines).
+Here's how it works: you'll setup a DVC remote with access credentials passed to
+your GitHub Action/GitLab CI via CML. Then you'll use the following protocol in
+your CI workflow (your workflow config file in GitHub/GitLab):
+
+```dvc
+dvc pull --run-cache
+dvc repro
+dvc push --run-cache
+```
+
+When you use this design, any artifacts of `dvc repro`, such as models or
+transformed datasets, will be saved in DVC storage and indexed by the pipeline
+version that generated them. You can access them via `dvc checkout` in your
```

Comment:
but first you would need to pull run cache locally as well ... also, does `dvc checkout` actually restores things from the run cache ... I thought only repro does that

Reply:
yep that's right. checkout only works if you've pulled locally.